src/reducers/executionsReducer.js

```diff
@@ -77,6 +79,14 @@ export default (state = INITIAL_STATE, { type, payload }) => {
           executions.findIndex(({ id: exId }) => exId === payload),
         ),
       );
+    case CREATE_VALIDATION_SUCCESS:
+      return state.update('executions', (executions) =>
+        payload.executions.reduce(
+          (executionsList, execution) =>
+            addExecutionToList(execution)(executionsList),
+          executions,
+        ),
```

Comment:
Would it possible to use `forEach` here? I feel that using `reduce` is a bit complicated, but maybe I'm missing something.

Reply:
A `forEach` would mean pushing sequentially each execution into the same executions list, which doesn't work in the immutable context (it returns an updated copy instead).

But you are right that `reduce` is too complicated there, I found `list.concat` is a lot more appropriate